be2-scala/src/main/scala/ch/epfl/pop/Server.scala

```diff
@@ -67,20 +68,24 @@ object Server {
           system.terminate()
           typedSystem.terminate()
       }
-
-      /* Server terminating logic */
-      StdIn.readLine // let it run until user presses return
-      bindingFuture
-        .flatMap(_.unbind()) // trigger unbinding from the port
-        .onComplete(_ => {
-          logger.info("Server terminated !")
-          system.terminate()
-          typedSystem.terminate()
-        }) // and shutdown when done
-
-      Behaviors.empty
+      /* Shutting down */
+      val shutdownListener = new Thread() {
+        override def run(): Unit = {
+          logger.warning("shutdown in 5s ");
+          try {
+            bindingFuture.flatMap(_.unbind()).onComplete(_ => { //trigger unbinding from the port
+                logger.info("Server terminated !")
+                system.terminate()
+                typedSystem.terminate()
+              }) // and shutdown when done
+          } catch {
+                case  e: InterruptedException =>   logger.warning("Server shutting thread was interrupted !")
+          }
+        }
+      };
+    Runtime.getRuntime().addShutdownHook(shutdownListener);
+    Behaviors.empty
```

Comment:
is the indent correct here? 🤔

Reply:
Does not seem like, however I am sure my IDE is configured with `.editorconfig  `